components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -937,58 +935,26 @@ void AdsServiceImpl::SetDebug() {
 }
 
 void AdsServiceImpl::ParseCommandLineSwitches() {
-  const base::CommandLine& command_line =
-      *base::CommandLine::ForCurrentProcess();
-  if (!command_line.HasSwitch(switches::kRewards)) {
-    return;
-  }
-
-  const std::string switch_value =
-      command_line.GetSwitchValueASCII(switches::kRewards);
-  ParseCommandLineRewardsSwitchValue(switch_value);
-}
-
-void AdsServiceImpl::ParseCommandLineRewardsSwitchValue(
-    const std::string& switch_value) {
-  if (switch_value.empty()) {
-    return;
-  }
-
-  const std::vector<std::string> flags =
-      base::SplitString(base::ToLowerASCII(switch_value), ",",
-                        base::KEEP_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
-
-  for (const auto& flag : flags) {
-    const std::vector<std::string> components = base::SplitString(
-        flag, "=", base::KEEP_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
-    if (components.size() != 2) {
-      continue;
-    }
+  using brave_rewards::RewardsFlags;
 
-    const std::string& name = components.at(0);
-    const std::string& value = components.at(1);
+  const auto& flags = RewardsFlags::ForCurrentProcess();
 
-    if (name == "staging") {
-      ads::mojom::Environment environment;
-
-      if (value == "true" || value == "1") {
+  if (flags.environment) {
+    ads::mojom::Environment environment;
+    switch (*flags.environment) {
```

Comment:
nit: `.value()` instead of `*`

Reply:
Is there a reason to prefer `value()` to the dereference operator here? (The dereference operator is used elsewhere in this function.)

Reply:
Fixed.